app/app.py

```diff
@@ -0,0 +1,76 @@
+"""
+Main training task.
+"""
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+
+import os, json, random
+
+from flask import Flask, render_template, jsonify, request, send_from_directory
+from annoy import AnnoyIndex
+
+
+### Util Functions
+def get_json(filename):
+    with open(filename) as json_data:
+        data = json.load(json_data)
+    return data
+
+def get_int(val):
+    try:
+        return int(val)
+    except (ValueError, TypeError):
+        return -1
+
+def validate_input(val):
+    val = get_int(val)
+    if val > -1 and val < limit:
+        return val
+    else:
+        return random.randrange(limit)
+
+def get_neighbors(image_id):
```

Comment:
Rather than using a global this should probably be a class.

```
class AnnoyLookup(object):

    def __init__(self, metadata_path='./data/metadata.json'):
        self._data = get_json(metadata_path)
        self._limit = len(self._data['filenames'])

        self._index = AnnoyIndex(self._data['features_length'])
        self._index.load('./data/index.ann')

    def get_neighbors(self, image_id, max_neighbors=13):
        results = []
        for item_id in self._index.get_nns_by_item(image_id, max_neighbors):
            results.append({
                'id': item_id,
                'image': self._data['filenames'][item_id]
            })
        return results
```